development/src/logic.ts

```diff
@@ -166,31 +166,36 @@ function actOnTurn(
     const targetSquare = target as Square;
     actOnTurnPieceToSquare(draggedPiece, targetSquare);
   }
-
-  endTurn();
 }
 
 function actOnTurnPieceToPiece(draggedPiece: Piece, targetPiece: Piece) {
-  Logger.log(
-    `A ${targetPiece.player.color} ${targetPiece.name} was killed by a ${draggedPiece.player.color} ${draggedPiece.name}.`,
-  );
-
-  isFriendlyFire = targetPiece.player === draggedPiece.player;
-
-  pieces = pieces.filter((piece) => piece !== targetPiece);
-  deathCounter++;
-  isPieceKilled = true;
-  destroyPieceOnBoard(targetPiece);
-
-  const targetSquare: Square = { position: targetPiece.position };
-  move(draggedPiece, targetSquare);
+  let isFriendlyFireCheck = targetPiece.player === draggedPiece.player;
+  let isSamePiece = targetPiece.position[0] === draggedPiece.position[0] &&
+   targetPiece.position[1] === draggedPiece.position[1];
+  if(isFriendlyFireCheck && !isSamePiece || !isFriendlyFireCheck)
+  {
+    isFriendlyFire = isFriendlyFireCheck;
+    pieces = pieces.filter((piece) => piece !== targetPiece);
+    deathCounter++;
+    isPieceKilled = true;
+    destroyPieceOnBoard(targetPiece);
+
+    const targetSquare: Square = { position: targetPiece.position };
+    move(draggedPiece, targetSquare);
+
+    Logger.log(
+      `A ${targetPiece.player.color} ${targetPiece.name} was killed by a ${draggedPiece.player.color} ${draggedPiece.name}.`,
+    );
+    endTurn();
+  }
```

Comment:
Wouldn't it make more sense to `if (draggedPiece === targetPiece) return;` after the log?